### PR TITLE
Update to use promises instead of callbacks

### PR DESCRIPTION
**Pull request by subpx**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -30,8 +30,7 @@ module.exports = function SvelteStateRendererFactory(defaultOptions = {}) {
 					svelte = construct(template.component, options)
 				}
 			} catch (e) {
-				cb(e)
-				return
+				throw new Error(e)
 			}
```
Comment (TehShrike) [inline index.js]: We should probably just delete the whole `catch` and let the original error bubble out as the rejection

File: index.js [hunk 2]
```diff
@@ -59,20 +58,21 @@ module.exports = function SvelteStateRendererFactory(defaultOptions = {}) {
...
-			getChildElement: function getChildElement(svelte, cb) {
+			getChildElement: async function getChildElement(svelte) {
 				try {
 					const element = svelte.mountedToTarget
 					const child = element.querySelector(`uiView`)
-					cb(null, child)
+					return child
 				} catch (e) {
-					cb(e)
+					throw new Error(e)
```
Comment (TehShrike) [inline index.js]: Same thing here, I think the try/catch can go away entirely

File: index.js [hunk 3]
```diff
@@ -59,20 +58,21 @@ module.exports = function SvelteStateRendererFactory(defaultOptions = {}) {
...
-				render(renderContext, cb)
+				return render(renderContext)
 			},
-			destroy: function destroy(svelte, cb) {
+			destroy: async function destroy(svelte) {
 				svelte.asrOnDestroy()
 				svelte.$destroy()
-				cb()
+
+				return
```
Comment (TehShrike) [inline index.js]:
```suggestion
```

**Review by TehShrike — COMMENTED** [review]
beep boop

**Comment by subpx** [comment]
Those changes are done

**Comment by TehShrike** [comment]
Published as 3.1.0

**Pull request merged**